src/CollectionsNg/Hamt.elm

```diff
@@ -419,7 +168,7 @@ set' shift hash key val ls =
                         newNodes =
                             nodes
                                 |> List.filter (\( k, _ ) -> k /= key)
-                                |> ((::) ( key, val ))
+                                |> ((++) [( key, val )])
```

Comment:
What's the thinking here? I think cons is faster than append (barring data to the contrary), and the list gets sorted the next line.